client/dashboard/app/auth/index.tsx

```diff
@@ -14,6 +15,27 @@ interface AuthContextType {
 }
 export const AuthContext = createContext< AuthContextType | undefined >( undefined );
 
+async function initializeCurrentUser(): Promise< User > {
+	// In support user session the `currentUser` refers to the wrong person so we should request
+	// the user object. Note we do not check `isSupportNextSession()` because in "next" support
+	// sessions the server does bootstrap the correct `currentUser`.
+	const useBootstrap = ! isSupportUserSession() && config.isEnabled( 'wpcom-user-bootstrap' );
+
+	if ( useBootstrap ) {
+		if ( window.currentUser ) {
+			// TODO: align the various `currentUser` types. The different types have
+			// different opinions on which fields are required and optional.
+			// - packages/api-core/src/me/types.ts
+			// - packages/data-stores/src/user/types.ts
+			// - client/lib/user/user.d.ts
+			return window.currentUser as any;
+		}
+		throw new Error( 'Failed to bootstrap user object' );
```

Comment:
Are you sure this won't break anonymous requests? Such a request should redirect to login, not display some 404 error.